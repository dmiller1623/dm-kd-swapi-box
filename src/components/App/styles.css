.app {
  display: flex;
  align-items: center; }

.header-parent {
  color: #f3f3f3;
  display: flex;
  justify-content: center; }

.button-parent {
  color: #f3f3f3;
  display: flex;
  justify-content: center; }

.movie-parent {
  display: flex;
  justify-content: center;
  height: 100vh;
  margin: 0;
  width: 25%; }

.card-container-parent {
  margin-top: 2rem;
  display: flex;
  justify-content: center; }

.main-section {
  width: 75%;
  height: 100vh;
  background-image: url(https://wallpaper-house.com/data/out/7/wallpaper2you_150538.png); }
