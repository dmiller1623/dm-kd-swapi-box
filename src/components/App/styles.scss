.App {
  display: flex;
  align-items: center;
}

.header-parent {
  display: flex;
  justify-content: center;
}

.button-parent {
  display: flex;
  justify-content: center;
}

.movie-parent {
  display: flex;
  justify-content: center;
  margin: 0;
  width: 20%;
}

.main-section {
  width: 80%;
}
