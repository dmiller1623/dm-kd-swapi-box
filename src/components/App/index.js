import React, { Component } from 'react';
import './styles.css';
import Header from '../Header';
import CardContainer from '../CardContainer'
import Button from '../Button'
import MovieText from '../MovieText';
// import { peopleData, planetData } from '../../helper';


class App extends Component {
  constructor () {
    super()
    this.state = ({
      numOfFaves: 0,
      movies: {},
      people: {},
      vehicles: {},
      planets: {}
    })
  }

  componentDidMount = () => {
    const movies = 'movies'
    const people = 'people'
    const planets = 'planets'
    const vehicles = 'vehicles'
    const peopleUrl = 'https://swapi.co/api/people/'
    const planetsUrl = 'https://swapi.co/api/planets/'
    const vehiclesUrl = 'https://swapi.co/api/vehicles/'
    const movieUrl = 'https://swapi.co/api/films/'
    

    // this.fetch(movies, movieUrl)
    // this.fetch(people, peopleUrl)
    // this.fetch(planets, planetsUrl)
    // this.fetch(vehicles, vehiclesUrl)

    this.setState( { movies: {title: 'Return of the Jedi', 
    opening_crawl: 'Luke Skywalker has returned to his home planet of Tatooine in an attempt to rescue his friend Han Solo from the clutches of the vile gangster Jabba the Hutt. Little does Luke know that the GALACTIC EMPIRE has secretly begun construction on a new armored space',
    release_date: '1983-05-25'} })
  }

  fetch = (dataType, url) => {
    fetch(url).then(response => response.json())
    .then(data => {
      console.log(dataType, data)
      if (dataType === 'movies') {
        this.setState({ [dataType]: data.results[4]})
      } else {
      this.setState({ [dataType]: data })
      }
    })
    .catch(error => {
    })
  }

  render() {
    return (
      <div className='app'>
        <div className='movie-parent'>
          <MovieText movies={this.state.movies}/>
        </div>
        <div className='main-section'>
          <div className='header-parent'>
            <Header numOfFaves={this.state.numOfFaves}/>
          </div>
          <div className='button-parent'>
            <Button />
          </div>
          <div className='card-container-parent'>
            <CardContainer />
          </div>
        </div>
      </div>
    );
  }
}

export default App;
