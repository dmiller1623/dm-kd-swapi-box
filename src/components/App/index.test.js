import React from 'react';
import ReactDOM from 'react-dom';
import App from './index.js';
import { shallow, mount } from 'enzyme'
import { spy } from 'sinon'

describe('App', () => {
  let wrapper
  beforeEach(() => {
    window.fetch = jest.fn().mockImplementation(() => 
    Promise.resolve({
      ok: true, 
      json: () => Promise.resolve({ name: 'human', population: 1 })
    }))
    wrapper = shallow(<App />)
    // const url = 'https://swapi.co/api/people/'
    // wrapper.instance().getMovieData(url)
  })
  
  it('should start off with a default state', () => {
    const expected = {
      numOfFaves: 0,
      movies: [],
      people: [],
      vehicles: [],
      planets: [],
      cardType: 'welcome',
      favorites: [],
      isLoading: false
    }
    expect(wrapper.state()).toEqual(expected)
    
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('movieData return the right data', async () => {
    const url = 'https://swapi.co/api/people/'
    wrapper.instance().getMovieData(url)
    expect(wrapper.state('movies')).toEqual(['hello'])
  })
})
