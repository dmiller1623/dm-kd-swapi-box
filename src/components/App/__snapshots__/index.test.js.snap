// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="movie-parent"
>
          <MovieText
                    movies={Array []}
          />
</div>,
        <div
          className="main-section"
>
          <div
                    className="header-parent"
          >
                    <Header
                              evaluateButtonClass={[Function]}
                              handleDisplayCards={[Function]}
                              numOfFaves={0}
                    />
          </div>
          <div
                    className="button-parent"
          >
                    <Button
                              evaluateButtonClass={[Function]}
                              handleDisplayCards={[Function]}
                    />
          </div>
          <div
                    className="card-container-parent"
          >
                    <h1
                              className="welcome-message"
                    >
                              Select a button from above
                    </h1>
                    <CardContainer
                              cardType="welcome"
                              evaluateClass={[Function]}
                              favorites={Array []}
                              people={Array []}
                              planets={Array []}
                              updateFavorites={[Function]}
                              vehicles={Array []}
                    />
          </div>
</div>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MovieText
            movies={Array []}
/>,
          "className": "movie-parent",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "movies": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="header-parent"
>
              <Header
                            evaluateButtonClass={[Function]}
                            handleDisplayCards={[Function]}
                            numOfFaves={0}
              />
</div>,
            <div
              className="button-parent"
>
              <Button
                            evaluateButtonClass={[Function]}
                            handleDisplayCards={[Function]}
              />
</div>,
            <div
              className="card-container-parent"
>
              <h1
                            className="welcome-message"
              >
                            Select a button from above
              </h1>
              <CardContainer
                            cardType="welcome"
                            evaluateClass={[Function]}
                            favorites={Array []}
                            people={Array []}
                            planets={Array []}
                            updateFavorites={[Function]}
                            vehicles={Array []}
              />
</div>,
          ],
          "className": "main-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Header
                evaluateButtonClass={[Function]}
                handleDisplayCards={[Function]}
                numOfFaves={0}
/>,
              "className": "header-parent",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "evaluateButtonClass": [Function],
                "handleDisplayCards": [Function],
                "numOfFaves": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Button
                evaluateButtonClass={[Function]}
                handleDisplayCards={[Function]}
/>,
              "className": "button-parent",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "evaluateButtonClass": [Function],
                "handleDisplayCards": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="welcome-message"
>
                  Select a button from above
</h1>,
                <CardContainer
                  cardType="welcome"
                  evaluateClass={[Function]}
                  favorites={Array []}
                  people={Array []}
                  planets={Array []}
                  updateFavorites={[Function]}
                  vehicles={Array []}
/>,
              ],
              "className": "card-container-parent",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Select a button from above",
                  "className": "welcome-message",
                },
                "ref": null,
                "rendered": "Select a button from above",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "cardType": "welcome",
                  "evaluateClass": [Function],
                  "favorites": Array [],
                  "people": Array [],
                  "planets": Array [],
                  "updateFavorites": [Function],
                  "vehicles": Array [],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="movie-parent"
>
            <MovieText
                        movies={Array []}
            />
</div>,
          <div
            className="main-section"
>
            <div
                        className="header-parent"
            >
                        <Header
                                    evaluateButtonClass={[Function]}
                                    handleDisplayCards={[Function]}
                                    numOfFaves={0}
                        />
            </div>
            <div
                        className="button-parent"
            >
                        <Button
                                    evaluateButtonClass={[Function]}
                                    handleDisplayCards={[Function]}
                        />
            </div>
            <div
                        className="card-container-parent"
            >
                        <h1
                                    className="welcome-message"
                        >
                                    Select a button from above
                        </h1>
                        <CardContainer
                                    cardType="welcome"
                                    evaluateClass={[Function]}
                                    favorites={Array []}
                                    people={Array []}
                                    planets={Array []}
                                    updateFavorites={[Function]}
                                    vehicles={Array []}
                        />
            </div>
</div>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MovieText
              movies={Array []}
/>,
            "className": "movie-parent",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "movies": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="header-parent"
>
                <Header
                                evaluateButtonClass={[Function]}
                                handleDisplayCards={[Function]}
                                numOfFaves={0}
                />
</div>,
              <div
                className="button-parent"
>
                <Button
                                evaluateButtonClass={[Function]}
                                handleDisplayCards={[Function]}
                />
</div>,
              <div
                className="card-container-parent"
>
                <h1
                                className="welcome-message"
                >
                                Select a button from above
                </h1>
                <CardContainer
                                cardType="welcome"
                                evaluateClass={[Function]}
                                favorites={Array []}
                                people={Array []}
                                planets={Array []}
                                updateFavorites={[Function]}
                                vehicles={Array []}
                />
</div>,
            ],
            "className": "main-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Header
                  evaluateButtonClass={[Function]}
                  handleDisplayCards={[Function]}
                  numOfFaves={0}
/>,
                "className": "header-parent",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "evaluateButtonClass": [Function],
                  "handleDisplayCards": [Function],
                  "numOfFaves": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Button
                  evaluateButtonClass={[Function]}
                  handleDisplayCards={[Function]}
/>,
                "className": "button-parent",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "evaluateButtonClass": [Function],
                  "handleDisplayCards": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="welcome-message"
>
                    Select a button from above
</h1>,
                  <CardContainer
                    cardType="welcome"
                    evaluateClass={[Function]}
                    favorites={Array []}
                    people={Array []}
                    planets={Array []}
                    updateFavorites={[Function]}
                    vehicles={Array []}
/>,
                ],
                "className": "card-container-parent",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Select a button from above",
                    "className": "welcome-message",
                  },
                  "ref": null,
                  "rendered": "Select a button from above",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "cardType": "welcome",
                    "evaluateClass": [Function],
                    "favorites": Array [],
                    "people": Array [],
                    "planets": Array [],
                    "updateFavorites": [Function],
                    "vehicles": Array [],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
